altair_pandas/_core.py

```diff
@@ -55,55 +53,54 @@ def scatter(self, **kwargs):
         raise ValueError("kind='scatter' can only be used for DataFrames.")
 
     def hist(self, **kwargs):
-        data = self._data.iloc[:, 1:]
+        data = self._preprocess_data(with_index=False)
         column = data.columns[0]
         return alt.Chart(data).mark_bar().encode(
             x=alt.X(column, title=None, bin=True),
             y=alt.Y('count()', title='Frequency'),
         )
 
     def box(self, **kwargs):
-        data = self._data.iloc[:, 1:]
+        data = self._preprocess_data(with_index=False)
         return alt.Chart(data).transform_fold(
             list(data.columns), as_=['column', 'value']
         ).mark_boxplot().encode(
```

Comment:
@jakevdp -- as a general pattern, should we stick to transforming data in vega or use `stack` and etc from pandas? If you prefer moving this to a issue discussion for larger community's benefit, we can do that.

Reply:
As a general pattern, I think transforms should happen in the begs transform layer as much as possible. I don't think any pandas-side preprocessing needs to happen beyond what's already defined here, which is the bear minimum required to make the data compatible with Altair.